codenames-domain/src/main/java/com/codenames/services/UserService.java

```diff
@@ -0,0 +1,79 @@
+package com.codenames.services;
+
+import com.codenames.dto.CredentialDto;
+import com.codenames.dto.SignUpDto;
+import com.codenames.entity.UserEntity;
+import com.codenames.exception.UserAlreadyExistsException;
+import com.codenames.exception.UserNotFoundException;
+import com.codenames.exception.UserParametersNoValidException;
+import com.codenames.mapper.UserMapper;
+import com.codenames.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+
+import java.nio.CharBuffer;
+import java.util.Optional;
+
+@Controller
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    private final PasswordEncoder passwordEncoder;
+
+    private final UserMapper userMapper;
+
+    public <R> R findByNickname(Class<R> rClass, String login){
```

Comment:
Same here

Reply:
So I reworked it. I named this generic because I meant it to be a type of return